app/customer.py

```diff
@@ -0,0 +1,66 @@
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list[int],
+            money: int | float
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.home_location = location
+        self.location = location
+        self.money = money
+        self.trip_costs = {}
+        self.chosen_shop = None
+        self.spent_money = 0
+
+    def amount_of_money_at_start(self) -> None:
+        print(f"{self.name} has {self.money} dollars")
+
+    def cost_of_all_trips(
+            self,
+            shops: list,
+            fuel_consumption: float,
+            fuel_price: float
+    ) -> None:
+        for shop in shops:
+            shop = Shop(
+                name=shop["name"],
+                location=shop["location"],
+                products=shop["products"]
+            )

```

Comment:
The instantiation of the 'Shop' class inside the loop may lead to unnecessary object creation if 'shop' is already an instance of 'Shop'. Consider checking if 'shop' is already an instance of 'Shop' before creating a new one to avoid redundancy.